### ISSUE TITLE
Batch Inferencing

### ISSUE THREAD

**Issue by nilesh23041999** [mlfoundations/open_clip#599]
How to batch inference to get 
image_features,  text_features.
Facing dimension issue

# Stack all the images into a single tensor
    image_tensors = torch.stack([preprocessor_openCLIP(img) for img in crop_imgs], dim=0)
    print('img shape batch ', image_tensors.shape)
    # Tokenize the query strings
    text_inputs = [tokenizer_openCLIP(query) for query in query_strings]
    print(text_inputs)
    max_length = max([input_tensor.size(0) for input_tensor in text_inputs])
    print(max_length)
    padded_text_inputs = pad_text_inputs(text_inputs, max_length)
    text_inputs_tensor = torch.stack(padded_text_inputs, dim=1)  # Convert to a tensor
    print('text shape batch ', text_inputs_tensor.shape)
    
    text_inputs_tensor_permuted = text_inputs_tensor.permute(1, 0, 2)

    with torch.no_grad(), torch.cuda.amp.autocast():
        
        image_features = model_openCLIP.encode_image(image_tensors)
        print(333, image_features.shape)
        text_features = model_openCLIP.encode_text(text_inputs_tensor_permuted)
        image_features /= image_features.norm(dim=-1, keepdim=True)

        text_features /= text_features.norm(dim=-1, keepdim=True)
        text_probs = (100.0 * image_features @ text_features.T).softmax(dim=-1)
    results_list = torch.flatten(text_probs, start_dim=0, end_dim=1).cpu().tolist()

**Comment by rwightman** [comment]
@nilesh23041999 the included tokenzers via the `tokenize()` fn, or the HfTokenizer (if using Hf tokenizer), already appropriately batches the inputs with padding if you pass lists of strings, so other than that, stacking same sized images on dim=0, everything will work..

https://github.com/mlfoundations/open_clip/blob/main/src/open_clip/tokenizer.py#L159-L214
